cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/configs/StorageProperties.java

```diff
@@ -0,0 +1,41 @@
+package com.linkedin.openhouse.cluster.storage.configs;
+
+import com.linkedin.openhouse.cluster.configs.YamlPropertySourceFactory;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.*;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+
+/**
+ * This class represents the storage properties for the cluster. It includes the default storage
+ * type and a map of different storage types. Each storage type has its own properties such as
+ * scheme, root path, endpoint, and parameters. For list of supported storage types, see {@link
+ * com.linkedin.openhouse.cluster.storage.StorageType}.
+ */
+@Configuration
+@ConfigurationProperties(prefix = "cluster.storages")
+@PropertySource(
+    name = "clusterStorage",
+    value = "file:${OPENHOUSE_CLUSTER_CONFIG_PATH:/var/config/cluster.yaml}",
+    factory = YamlPropertySourceFactory.class,
+    ignoreResourceNotFound = true)
+@Getter
+@Setter
+public class StorageProperties {
+  private String defaultType;
+  private Map<String, StorageTypeProperties> types;
```

Comment:
We need a validation method here, given there could be misconfigurations or typo. We could start with some basic validations, for example, we could make sure if the storage type is hdfs, then the endpoint startsWith("hdfs:///"), and so on and so forth.

Reply:
Good point! we'll add it when these properties get used to construct beans. The goal of this property is simply to be a POJO for yaml confs. If they are misconfigured, we can make a decision when they get used.